core/src/Kokkos_ScratchSpace.hpp

```diff
@@ -31,103 +31,77 @@ static_assert(false,
 
 namespace Kokkos {
 
-/** \brief  Scratch memory space associated with an execution space.
- *
- */
-template <class ExecSpace>
-class ScratchMemorySpace {
+struct L0Tag {};
+
+struct L1Tag {};
```

Comment:
maybe we should make the name hint that they are used for `Scratch`? like `ScratchL0Tag`?

Reply:
Sure, those names a re up for discussion. I would also be fine with `CacheScratchTag` and `GlobalScratchTag` or similar. Ultimately it should match the name for the alias used for the specialized spaces.
`level=0` and` level=1` is just what we are currently using. 
Keeping integers for the public interfaces for requesting a space is convenient for transitioning but we could use these tags there as well.